staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -228,7 +228,11 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	case len(subresource) == 0:
 		handler = r.serveResource(w, req, requestInfo, crdInfo, terminating, supportedTypes)
 	default:
-		http.Error(w, "the server could not find the requested resource", http.StatusNotFound)
+		msg := resource
+		if len(subresource) != 0 {
+			msg = fmt.Sprintf("%s/%s", resource, subresource)
+		}
+		http.Error(w, fmt.Sprintf("the server could not find the requested resource: %s", msg), http.StatusNotFound)
```

Comment:
I wonder, is it common to have part of the http path (which this essentially is) in the error messages? What do we do elsewhere?

Reply:
> I wonder, is it common to have part of the http path (which this essentially is) in the error messages?

not really... 

> What do we do elsewhere?

if you request an unknown group, or an unknown resource inside a known native group, you get `{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the server could not find the requested resource","reason":"NotFound","details":{},"code":404}`